### PR TITLE
SAP - Ignore already deleted lease

### PR DESCRIPTION
**Pull request by hemna**
When we try and release a lease, we might get an object not found
exception meaning that the lease doesn't exist.  So there is no need
to release something that doesn't exist.

### CODE REVIEW

File: oslo_vmware/rw_handles.py [hunk 1]
```diff
@@ -558,6 +560,10 @@ def close(self):
         """
         try:
             self._release_lease()
+        except exceptions.ManagedObjectNotFoundException:
```
Comment (mariusleu) [inline oslo_vmware/rw_handles.py:563]: This extends `exceptions.VimException`, so the below `except` would catch this as well.
Is this change then only for having a different log message?

File: oslo_vmware/rw_handles.py [hunk 2]
```diff
@@ -558,6 +560,10 @@ def close(self):
...
+            LOG.info("Lease for %(url)s not found.  No need to release.",
+                     {'url': self._url})
+            return
```
Comment (mariusleu) [inline oslo_vmware/rw_handles.py:566]: You exit here, but there is an extra `close()` call at the end of this function, which would close the http connection.

**Comment by hemna** [comment]
Ok so it looks like there is a discrepancy between the read and write handles on how they handle the broken lease.
The read handle reraises when the lease is gone, but the write handle doesn't?
I'll make them both the same.   with a finally clause and a re-raise.

**Review by mariusleu — APPROVED** [review]

**Comment by hemna** [comment]
fwiw, the unit tests passes in py36

**Pull request merged**